README.md
# onboarding slider
This is a package that extends the sk_onboarding_screen with a few extra options.  

A library that walks a user through multiple on-boarding screens in a simple and easy way
  
 [![Flutter](https://img.shields.io/badge/Platform-Flutter-blue.svg)](https://flutter.dev/)

## 💻 Installation

You just need to add `onboarding_slider` as a [dependency in your pubspec.yaml file.](https://flutter.dev/docs/development/packages-and-plugins/using-packages)

```yaml
dependencies:
onboarding_slider: ^1.0.0
```

## Usage

### Import this class

```dart
import 'package:onboarding_slider/flutter_onboarding.dart';
import 'package:onboarding_slider/onboarding_slider_model.dart';
```

## OnboardingSliderModel

```dart
  final pages = [
    OnboardingSliderModel(
        title: 'Choose your item',
        description:
            'Easily find your grocery items and you will get delivery in wide range',
        titleColor: Colors.black,
        descripColor: const Color(0xFF929794),
        imagePath: 'assets/onboarding1.png'),
    OnboardingSliderModel(
        title: 'Pick Up or Delivery',
        description:
            'We make ordering fast, simple and free-no matter if you order online or cash',
        titleColor: Colors.black,
        descripColor: const Color(0xFF929794),
        imagePath: 'assets/onboarding2.png'),
    OnboardingSliderModel(
        title: 'Pay quick and easy',
        description: 'Pay for order using credit or debit card',
        titleColor: Colors.black,
        descripColor: const Color(0xFF929794),
        imagePath: 'assets/onboarding3.png'),
  ];
```
### Pass it into OnboardingSliderScreen Widget

```dart
  @override
  Widget build(BuildContext context) {
    // TODO: implement build
    return Scaffold(
      body: OnboardingSliderScreen(
        bgColor: Colors.white,
        loopPages: true,
        showSkip: true,
        showgetStarted : true,
        loopDuration :const Duration(seconds:1),
        themeColor: const Color(0xFFf74269),
        pages: pages,
        skipClicked: (value) {
          print("Skip");
        },
        getStartedClicked: (value) {
          print("Get Started");
        },
      ),
    );
  }
```

## 📃License

    Copyright 2020

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.


CHANGELOG.md
## [0.0.2] - 2020-12-17
* Support for SVG

## [0.0.1] - 2020-12-14
* First Version


lib/onboarding_slider_screen.dart
library sk_onboarding_screen;

import 'dart:async';

import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_svg/svg.dart';
import 'package:onboarding_slider/onboarding_slider_model.dart';

/// A library that walks a user through multiple on-boarding screens in a simple and easy way
class OnboardingSliderScreen extends StatefulWidget {
  ///  A list of pages to be displayed
  final List<OnboardingSliderModel> pages;

  /// Set the background color
  final Color bgColor;

  /// Also set a theme color
  final Color themeColor;

  /// Should the pages automatically loop
  /// Default true
  final bool loopPages;

  /// If pages should loop
  /// set a duration for the loop
  final Duration loopDuration;

  /// should the skip button be shown
  /// Default is true
  final bool showSkip;

  /// Enter the skip action
  final ValueChanged<String> skipClicked;

  /// should the getstarted button be shown
  /// default is true
  final bool showgetStarted;

  /// getstarted button action
  final ValueChanged<String> getStartedClicked;

  /// A library that walks a user through multiple on-boarding screens in a simple and easy way
  OnboardingSliderScreen({
    Key key,
    @required this.pages,
    @required this.bgColor,
    @required this.themeColor,
    this.loopDuration = const Duration(seconds: 5),
    this.loopPages = true,
    this.skipClicked,
    this.showSkip = true,
    this.getStartedClicked,
    this.showgetStarted = true,
  }) : super(key: key);

  @override
  OnboardingSliderScreenState createState() => OnboardingSliderScreenState();
}

class OnboardingSliderScreenState extends State<OnboardingSliderScreen> {
  final PageController _pageController = PageController(initialPage: 0);
  Timer _timer;
  int _currentPage = 0;
  bool _autochanging = false;
  List<Widget> _buildPageIndicator() {
    List<Widget> list = [];
    for (int i = 0; i < widget.pages.length; i++) {
      list.add(i == _currentPage ? _indicator(true) : _indicator(false));
    }
    return list;
  }

  List<Widget> buildOnboardingPages() {
    final children = <Widget>[];

    for (int i = 0; i < widget.pages.length; i++) {
      children.add(_showPageData(widget.pages[i]));
    }
    return children;
  }

  @override
  void initState() {
    // TODO: implement initState
    super.initState();
    //start timer
    if (widget.loopPages) {
      _timer = new Timer.periodic(
        widget.loopDuration,
        (Timer timer) {
          changeSlides();
        },
      );
    }
  }

  @override
  void dispose() {
    if (widget.loopPages) {
      _timer.cancel();
    }
    super.dispose();
  }

  void changeSlides() {
    _autochanging = true;
    if (_currentPage == widget.pages.length - 1) {
      _currentPage = 0;
      _pageController.animateToPage(0,
          duration: Duration(milliseconds: 500), curve: Curves.ease);
    } else {
      _currentPage += 1;
      _pageController.nextPage(
        duration: Duration(milliseconds: 500),
        curve: Curves.ease,
      );
    }
    setState(() {});
  }

  Widget _indicator(bool isActive) {
    return AnimatedContainer(
      duration: Duration(milliseconds: 150),
      margin: EdgeInsets.symmetric(horizontal: 8.0),
      height: 10.0,
      width: isActive ? 15.0 : 10.0,
      decoration: BoxDecoration(
        color: isActive ? widget.themeColor : Color(0xFF929794),
        borderRadius: BorderRadius.all(Radius.circular(10)),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: widget.bgColor,
      body: AnnotatedRegion<SystemUiOverlayStyle>(
        value: SystemUiOverlayStyle.light,
        child: SafeArea(
          child: Container(
            height: MediaQuery.of(context).size.height,
            child: Padding(
              padding: EdgeInsets.symmetric(vertical: 10.0),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.stretch,
                children: <Widget>[
                  widget.showSkip
                      ? Container(
                          alignment: Alignment.centerRight,
                          child: FlatButton(
                            onPressed: () {
                              widget.skipClicked("Skip Tapped");
                            },
                            child: Text(
                              'Skip',
                              style: TextStyle(
                                fontWeight: FontWeight.w500,
                                color: Colors.black,
                                fontSize: 20,
                              ),
                            ),
                          ),
                        )
                      : Container(),
                  Container(
                    height: 500.0,
                    color: Colors.transparent,
                    child: PageView(
                        physics: ClampingScrollPhysics(),
                        controller: _pageController,
                        onPageChanged: (int page) {
                          //to differentiate manual scroll
                          if (_autochanging == false) {
                            if (widget.loopPages) {
                              _timer.cancel();
                            }
                          }
                          _autochanging = false;
                          setState(() {
                            _currentPage = page;
                          });
                        },
                        children: buildOnboardingPages()),
                  ),
                  Row(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: _buildPageIndicator(),
                  ),
                  // _currentPage != widget.pages.length - 1
                  //     ? Expanded(
                  //         child: Align(
                  //           alignment: FractionalOffset.bottomRight,
                  //           child: Padding(
                  //             padding: EdgeInsets.only(right: 20, bottom: 10),
                  //             child: FloatingActionButton(
                  //               backgroundColor: widget.bgColor,
                  //               child: Icon(
                  //                 Icons.arrow_forward,
                  //                 color: widget.themeColor,
                  //               ),
                  //               onPressed: () {
                  //                 _pageController.nextPage(
                  //                   duration: Duration(milliseconds: 500),
                  //                   curve: Curves.ease,
                  //                 );
                  //               },
                  //             ),
                  //           ),
                  //         ),
                  //       )
                  //     : Text(''),
                ],
              ),
            ),
          ),
        ),
      ),
      bottomSheet: (_currentPage == widget.pages.length - 1) &&
              widget.showgetStarted == true
          ? _showGetStartedButton()
          : Text(''),
    );
  }

  Widget _showPageData(OnboardingSliderModel page) {
    return Padding(
      padding: EdgeInsets.all(40.0),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: <Widget>[
          (page.svgPath != null && page.svgPath == "")
              ? page.imagePath != null && page.imagePath == ""
                  ? Container()
                  : Center(
                      child: Image(
                        image: AssetImage(page.imagePath),
                        height: 300.0,
                        width: 300.0,
                      ),
                    )
              : Center(
                  child: SvgPicture.asset(
                    page.svgPath,
                    height: 300.0,
                    width: 300.0,
                  ),
                ),
          SizedBox(height: 30.0),
          Text(
            page.title,
            style: TextStyle(
              fontWeight: FontWeight.w700,
              color: page.titleColor,
              fontSize: 20,
            ),
          ),
          SizedBox(height: 15.0),
          Text(
            page.description,
            style: TextStyle(
              fontWeight: FontWeight.w400,
              color: page.descripColor,
              fontSize: 16,
            ),
          ),
        ],
      ),
    );
  }

  Widget _showGetStartedButton() {
    // if(widget.showgetStarted == false){
    //   return Container();
    // }
    final GestureDetector loginButtonWithGesture = new GestureDetector(
      onTap: _getStartedTapped,
      child: new Container(
        height: 50.0,
        decoration: new BoxDecoration(
            color: widget.themeColor,
            borderRadius: new BorderRadius.all(Radius.circular(6.0))),
        child: new Center(
          child: new Text(
            'Get Started',
            style: new TextStyle(
                color: Colors.white,
                fontSize: 20.0,
                fontWeight: FontWeight.w500),
          ),
        ),
      ),
    );

    return new Padding(
        padding:
            EdgeInsets.only(left: 20.0, right: 20.0, top: 5.0, bottom: 30.0),
        child: loginButtonWithGesture);
  }

  void _getStartedTapped() {
    widget.getStartedClicked("Get Started Tapped");
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:onboarding_slider/flutter_onboarding.dart';
import 'package:onboarding_slider/onboarding_slider_model.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(debugShowCheckedModeBanner: false, home: HomePage());
  }
}

class HomePage extends StatefulWidget {
  @override
  State<StatefulWidget> createState() {
    return HomePageState();
  }
}

class HomePageState extends State<HomePage> with TickerProviderStateMixin {
  final GlobalKey<ScaffoldState> _globalKey = new GlobalKey<ScaffoldState>();

  @override
  Widget build(BuildContext context) {
    // TODO: implement build
    return Scaffold(
      key: _globalKey,
      body: OnboardingSliderScreen(
        bgColor: Colors.white,
        loopPages: true,
        showSkip: true,
        showgetStarted: true,
        loopDuration: const Duration(seconds: 1),
        themeColor: const Color(0xFFf74269),
        pages: pages,
        skipClicked: (value) {
          print(value);
          _globalKey.currentState.showSnackBar(SnackBar(
            content: Text("Skip clicked"),
          ));
        },
        getStartedClicked: (value) {
          print(value);
          _globalKey.currentState.showSnackBar(SnackBar(
            content: Text("Get Started clicked"),
          ));
        },
      ),
    );
  }

  final pages = [
    OnboardingSliderModel(
        title: 'Choose your item',
        description:
            'Easily find your grocery items and you will get delivery in wide range',
        titleColor: Colors.black,
        descripColor: const Color(0xFF929794),
        imagePath: null,
        svgPath: 'assets/slider1.svg'),
    OnboardingSliderModel(
        title: 'Pick Up or Delivery',
        description:
            'We make ordering fast, simple and free-no matter if you order online or cash',
        titleColor: Colors.black,
        descripColor: const Color(0xFF929794),
        imagePath: 'assets/onboarding2.png',
        svgPath: ""),
    OnboardingSliderModel(
        title: 'Pay quick and easy',
        description: 'Pay for order using credit or debit card',
        titleColor: Colors.black,
        descripColor: const Color(0xFF929794),
        imagePath: "",
        svgPath: ""),
  ];
}
